src/Uno.UI/UI/Xaml/DependencyObjectStore.cs

```diff
@@ -1634,13 +1634,8 @@ internal IEnumerable<ResourceDictionary> GetResourceDictionaries(
 					{
 						yield return fe.Resources;
 					}
-
-					candidate = fe.Parent as FrameworkElement;
-				}
-				else
-				{
-					candidate = VisualTreeHelper.GetParent(candidate) as DependencyObject;
 				}
+				candidate = VisualTreeHelper.GetParent(candidate) as DependencyObject;
```

Comment:
i think we should keep the fe.Parent line for performance in general, since this will get spammed a lot
if i had to guess, the problem in the associated issue is in the native component.
logical tree: Page[contains style] > SV[also contains style] > StackPanel > Button
visual tree: Page[contains style] > page-template-stuffs > SV[also contains style] > ios-sv-template-stuffs > StackPanel > Button
and we have native views in "ios-sv-template-stuffs", that is weaved between FEs

so we can do something like: // chains `as DO` if really needed
```suggestion
				candidate =
					(candidate as FrameworkElement)?.Parent ??
					VisualTreeHelper.GetParent(candidate);
```

Reply:
actually i might be wrong here since if that were the case, we should flow through StackPanel -> SV directly without any problem
it might be useful to see difference in the hierarchy when navigating up using old code vs your new code, and maybe mine